src/testWithSpringBoot_3_4/java/org/openrewrite/java/spring/boot3/SpringBootVersionUpgradeTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.spring.boot3;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.config.Environment;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.openrewrite.java.Assertions.mavenProject;
+import static org.openrewrite.maven.Assertions.pomXml;
+
+class SpringBootVersionUpgradeTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec
+          .expectedCyclesThatMakeChanges(2)
+          .recipe(Environment.builder()
+            .scanRuntimeClasspath("org.openrewrite.java.spring")
+            .build()
+            .activateRecipes("org.openrewrite.java.spring.boot3.UpgradeSpringBoot_3_4")
+          );
+    }
+
+    @Test
+    @DocumentExample
+    void upgradeVersion() {
+        rewriteRun(
+          mavenProject("project",
+            //language=xml
+            pomXml(
+              """
+                <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+                    <modelVersion>4.0.0</modelVersion>
+                    <groupId>com.example</groupId>
+                    <artifactId>fooservice</artifactId>
+                    <version>1.0-SNAPSHOT</version>
+                    <parent>
+                        <groupId>org.springframework.boot</groupId>
+                        <artifactId>spring-boot-starter-parent</artifactId>
+                        <version>2.2.2.RELEASE</version>
+                        <relativePath/>
+                    </parent>
+                </project>
+                """,
+              spec -> spec.after(actual -> {
+                  assertThat(actual).containsPattern("<version>3.3.\\d+</version>");
```

Comment:
Shouldn't this be 3.4?

Reply:
Only once that version is released this Thursday: https://spring.io/projects#release-calendar

Reply:
FIxed in https://github.com/openrewrite/rewrite-spring/pull/629/commits/051a5768e0e9ee9abd42c789d18e2721e3ce461c